### PR TITLE
HDDS-9136. Throw exception when rename fails during moveToTrash.

### PR DESCRIPTION
**Pull request by ashishkumar50**
## What changes were proposed in this pull request?

moveToTrash is not checking for return value of rename API. When rename fails due to various reasons, still moveToTrash assume it as success. This leads to inconsistent behaviour, one which is mentioned in Jira description(HDDS-9136).
In this PR, If there is failure in rename operation, we are throwing exception to know moveToTrash has failed. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9136

## How was this patch tested?

Manually tested.

### CODE REVIEW

**Comment by ashishkumar50** [comment]
@sadanand48 @jojochuang Please help to review, thanks.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java [hunk 1]
```diff
@@ -338,7 +338,19 @@ private class RenameIterator extends OzoneListingIterator {
     boolean processKeyPath(List<String> keyPathList) throws IOException {
       for (String keyPath : keyPathList) {
         String newPath = dstPath.concat(keyPath.substring(srcPath.length()));
-        adapterImpl.rename(this.bucket, keyPath, newPath);
+        try {
+          adapterImpl.rename(this.bucket, keyPath, newPath);
```
[the remark this answers is not in the document]
Comment (ashishkumar50) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java:342]: Done

**Review by jojochuang — COMMENTED** [review]
wait for the precommit test. I suspect this change'll break the contract tests.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java [hunk 2]
```diff
@@ -350,7 +350,19 @@ boolean processKey(List<String> keyList) throws IOException {
       // TODO RenameKey needs to be changed to batch operation
       for (String key : keyList) {
         String newKeyName = dstKey.concat(key.substring(srcKey.length()));
-        adapter.renameKey(key, newKeyName);
+        try {
+          adapter.renameKey(key, newKeyName);
+        } catch (OMException ome) {
+          LOG.error("rename key failed: {}. source:{}, destin:{}",
```
Comment (hemantk-12) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:356]: nit: I think the log message can be better.
```
LOG.error("Key rename failed for source key: {} to " +
        "destination key: {}.", key, newKeyName, ome);
``` 
or:
```
LOG.error("Key rename failed with error: {}. Source key: {}, " +
        "Destination key: {}", ome.getMessage(), key, newKeyName);
```

Same for https://github.com/apache/ozone/pull/5253/files#diff-b2802792bb08b569714c90f10ecda42d917dd6329137188472010d2ea7476b1bR344

**Review by hemantk-12 — COMMENTED** [review]
Thanks for the patch @ashishkumar50

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java (hunk 2 above)
Comment (ashishkumar50) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java]: Done.

**Review by hemantk-12 — COMMENTED** [review]
Overall changes look good to me.

Is it possible to have some kind of unit or integration test for it? If yes, please add them.

**Comment by ashishkumar50** [comment]
> Overall changes look good to me.
> 
> Is it possible to have some kind of unit or integration test for it? If yes, please add them.

I tried but it's difficult to have a test for this.

**Review by ChenSammi — APPROVED** [review]

**Pull request merged**